addons/website_slides/models/slide_slide.py

```diff
@@ -401,29 +503,29 @@ def create(self, values):
             # 'website_published' is handled by mixin
             values['date_published'] = False
 
-        if values.get('slide_category') == 'infographic' and not values.get('image_1920'):
-            values['image_1920'] = values['binary_content']
         if values.get('is_category'):
             values['is_preview'] = True
             values['is_published'] = True
         if values.get('is_published') and not values.get('date_published'):
             values['date_published'] = datetime.datetime.now()
-        if values.get('url') and not values.get('document_id'):
-            doc_data = self._parse_document_url(values['url']).get('values', dict())
-            for key, value in doc_data.items():
-                values.setdefault(key, value)
 
         slide = super(Slide, self).create(values)
 
+        # avoid fetching external metadata when installing the module (i.e. for demo data)
+        # we also support a context key if you don't want to fetch the metadata when creating a slide
+        if any(values.get(url_param) for url_param in ['url', 'video_url', 'document_google_url', 'image_google_url']) \
+           and not self.env.context.get('install_mode', False) \
+           and self.env.context.get('website_slides_fetch_metadata', True):
+            slide_metadata = slide._fetch_external_metadata()
+            if slide_metadata:
+                # only update keys that are not set in the incoming values
+                slide.update({key: value for key, value in slide_metadata.items() if key not in values.keys()})
```

Comment:
Why not calling ``_on_change_url`` manually ? Something like
```
if not self.env.context.get('install_mode') and not
    self.env.context.get('website_slides_skip_fetch_metadata):
    slide._on_change_url()
```

That way we rely on the onchange and a bit less on "call method and then do an update".

Edit: actually you filter out values not already given at create, ok. Then maybe call ``_fetch_external_metadata`` only if we have values to update, to avoid an unnecessary call. If everything is given, hop hop skip it.

Reply:
That would actually be a bit complicated because the fact that "we have all the information" depends on the slide_category.

So I would have to do:
If it's a video and we have title / image / slide_type / duration in values then don't fetch.
If it's an image and we have title / slide_type in values then don't fetch.
etc...
Because if only one of those is missing, then we want to complete the data.

I'm thinking that the context key will be more clear in the end.
Also, I know that it's not a strong argument, but the previous code worked the same way, it fetched the metadata without checking if we have all the information.

If you really think we need to change this, I can give it a try in a separate commit to see how it looks.
Will have to re-test most flows as well.